app/metal-controller-manager/internal/server/server.go

```diff
@@ -258,7 +273,89 @@ func (s *server) Heartbeat(ctx context.Context, in *api.HeartbeatRequest) (*api.
 	return resp, nil
 }
 
-func Serve(c controllerclient.Client, recorder record.EventRecorder, scheme *runtime.Scheme, autoAccept, insecureWipe bool, rebootTimeout time.Duration) error {
+func (s *server) UpdateBMCInfo(ctx context.Context, in *api.UpdateBMCInfoRequest) (*api.UpdateBMCInfoResponse, error) {
+	if in.GetBmcInfo() != nil {
+		// Create or update creds secret
+		credsSecret := &corev1.Secret{}
+		exists := true
+
+		// Fetch corresponding server
+		obj := &metalv1alpha1.Server{}
+
+		if err := s.c.Get(ctx, types.NamespacedName{Name: in.GetUuid()}, obj); err != nil {
+			return nil, err
+		}
+
+		// For auto-created BMC info, we will *always* drop the creds in sidero-system namespace
+		if err := s.c.Get(ctx, types.NamespacedName{Namespace: "sidero-system", Name: in.GetUuid() + "-bmc"}, credsSecret); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, err
+			}
+
+			log.Printf("BMC secret doesn't exist for server %q, creating.", in.GetUuid())
+
+			credsSecret.ObjectMeta = metav1.ObjectMeta{
+				Namespace: "sidero-system",
+				Name:      in.GetUuid() + "-bmc",
+				OwnerReferences: []metav1.OwnerReference{
```

Comment:
Does clusterctl move move this resource?

Reply:
They should indeed get moved, but there's some nuance to it since secrets are namespaced. I'm going to move back to creating these secrets in the default namespace since that's where our clusters are also created by default and thus the secrets would get picked up properly with a `clusterctl move`